### ISSUE TITLE
different prediction results for pytorch 1.8.0 and 1.1.0 with the same model weight

### ISSUE THREAD

**Issue by lauraset** [pytorch/pytorch#53873]
## 🐛 Bug

When i load the same model weight in the pytorch 1.8.0 and 1.1.0,  large differences occur between the two results.
## To Reproduce

1. configure two environments: pytorch 1.8.0 and pytorch 1.1.0
1. load the same weight file, for example, model1.ckpt
1. set eval mode (i.e., model.eval()) to test the same images

## Expected behavior
results from the two environments should be similar in theory, but large differences were observed.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).
result from the pytorch 1.8.0
![image](https://user-images.githubusercontent.com/39206462/110890294-b694c400-832a-11eb-963a-58f616d75e9f.png)
result from the pytorch 1.1.0
![image](https://user-images.githubusercontent.com/39206462/110890344-d5935600-832a-11eb-9157-0ae50f70e355.png)

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.8.0 and 1.1.0
 - OS (e.g., Linux): windows 10
 - How you installed PyTorch (`conda`, `pip`, source): for both pytorch versions, I use "pip"
 - Build command you used (if compiling from source):
 - Python version: python 3.6
 - CUDA/cuDNN version: for 1.8.0, I use CUDA 11.1; for 1.1.0, I use CUDA 10.0
 - GPU models and configuration: for 1.8.0, I use  GeForce RTX 3090; for 1.1.0, I use GeForce GTX 1080
 - Any other relevant information:

**Comment by xwang233** [comment]
Can you share your training script?

**Comment by albanD** [comment]
Hi,

Unfortunately, there is no way we can guarantee reproducibility across different versions of CUDA, cudnn or PyTorch.
You can check the note on reproducibility here: https://pytorch.org/docs/stable/notes/randomness.html

The problem you see is that some low level lib is returning a result slightly different ( by maybe 1bit) and then your training is amplifying this difference as it goes. Until it becomes noticeable.
Note that this is usually due to the fact that the original model was not stable and would be failing in a similar manner (with the old version of libs) when using different random initialization.

**Comment by lauraset** [comment]
> Hi,
> 
> Unfortunately, there is no way we can guarantee reproducibility across different versions of CUDA, cudnn or PyTorch.
> You can check the note on reproducibility here: https://pytorch.org/docs/stable/notes/randomness.html
> 
> The problem you see is that some low level lib is returning a result slightly different ( by maybe 1bit) and then your training is amplifying this difference as it goes. Until it becomes noticeable.
> Note that this is usually due to the fact that the original model was not stable and would be failing in a similar manner (with the old version of libs) when using different random initialization.

Thank you very much
